src/System.IO.FileSystem/src/System/IO/Enumeration/FileSystemEnumerator.Windows.cs

```diff
@@ -118,6 +118,17 @@ private IntPtr CreateDirectoryHandle(string path)
             return handle;
         }
 
+        private bool ContinueOnDirectoryError(int error, bool ignoreNotFound)
+        {
+            // Directories can be removed (ERROR_FILE_NOT_FOUND) or replaced with a file of the same name (ERROR_DIRECTORY) while
+            // we are enumerating. The only reasonable way to handle this is to simply move on. There is no such thing as a "true"
+            // snapshot of filesystem state- our "snapshot" will consider the name non-existent in this rare case.
+
+            return (ignoreNotFound && (error == Interop.Errors.ERROR_FILE_NOT_FOUND || error == Interop.Errors.ERROR_PATH_NOT_FOUND || error == Interop.Errors.ERROR_DIRECTORY))
+                || (error == Interop.Errors.ERROR_ACCESS_DENIED && _options.IgnoreInaccessible)
+                || ContinueOnError(error);
```

Comment:
We seem to be inconsistent about when we call ContinueOnError in relation to other checks.  Here, we might not call ContinueOnError if one of the earlier conditions is met.  But here for example https://github.com/dotnet/corefx/pull/29908/files#diff-cbe887baa737b69109146305bbf6ca30R85, we call it first.

Reply:
Good catch. The design is to only give the error if the user code can make a decision about it.